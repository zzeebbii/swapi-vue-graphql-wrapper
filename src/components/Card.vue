<template>
  <div class="card">
    <div class="title">
      <h2>{{ title }}</h2>
    </div>
    <div class="info">
      <h3>{{ info }}</h3>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Card',
  props: {
    title: String,
    info: String
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.card {
  display: flex;
  flex-direction: column;
  width: 300px;
  border: 1px solid #42b983;
  border-radius: 5px;
  margin: 10px;
  cursor: pointer;
  text-align: center;

  .title {
    border-bottom: 1px solid #42b983;
    padding: 0.5rem;

    h1 {
      margin: 0;
    }
  }

  .info {
    padding: 0.5rem;
  }
}
</style>
