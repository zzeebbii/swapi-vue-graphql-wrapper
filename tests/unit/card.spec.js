import { shallowMount } from "@vue/test-utils";
import Card from "@/components/Card.vue";

describe("Card.vue", () => {
  it("renders props when passed", () => {
    const title = "Test Title";
    const info = "Test Info";
    const wrapper = shallowMount(Card, {
      propsData: { title, info }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
