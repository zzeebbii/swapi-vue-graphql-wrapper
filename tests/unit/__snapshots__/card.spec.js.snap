// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue renders props when passed 1`] = `
<div class="card">
  <div class="title">
    <h2>Test Title</h2>
  </div>
  <div class="info">
    <h3>Test Info</h3>
  </div>
</div>
`;
